File: demo/js/full_demo.js
import { serializePayload } from "../../src/zk/serialization.js";
import { hashPayload } from "../../src/zk/hashing.js";
import { Orchestrator } from "../../src/orchestrator/orchestrator.js";
import { SyncEngine } from "../../src/sync/sync_engine.js";
import { Aggregator } from "../../src/aggregation/aggregation.js";

async function runDemo() {
  const input = { user: "demo", progress: 42 };

  const serialized = serializePayload(input);
  const hash = hashPayload(serialized);

  const orch = new Orchestrator();
  const sync = new SyncEngine();
  const agg = new Aggregator();

  const step1 = orch.process(serialized);
  const step2 = sync.sync(step1);
  const final = agg.aggregate([step1, step2], hash);

  console.log("FINAL OUTPUT:", final);
}

runDemo();

File: demo/python/full_demo.py
from src.zk.serialization import serialize_payload
from src.zk.hashing import hash_payload
from src.orchestrator.orchestrator import Orchestrator
from src.sync.sync_engine import SyncEngine
from src.aggregation.aggregation import Aggregator

def run_demo():
    input_data = {"user": "demo", "progress": 42}

    serialized = serialize_payload(input_data)
    hashed = hash_payload(serialized)

    orch = Orchestrator()
    sync = SyncEngine()
    agg = Aggregator()

    step1 = orch.process(serialized)
    step2 = sync.sync(step1)
    final = agg.aggregate([step1, step2], hashed)

    print("FINAL OUTPUT:", final)

if __name__ == "__main__":
    run_demo()
